### ISSUE TITLE
`make test` failed after run `bin/gazelle`

### ISSUE THREAD

**Issue by gyliu513** [istio/mixer#1116]
After run `bin/gazelle`, then `make test`, some cases will be failed.

```
root@pc002:~/go/src/istio.io/mixer# git status
On branch build
Your branch is up-to-date with 'upstream/master'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   adapter/BUILD
	modified:   adapter/denier/BUILD
	modified:   adapter/list/BUILD
	modified:   adapter/noop2/BUILD
	modified:   pkg/adapter/BUILD
	modified:   pkg/aspect/BUILD
	modified:   pkg/config/BUILD
	modified:   pkg/template/BUILD
	modified:   template/BUILD
	modified:   template/sample/BUILD

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	diff
	pkg/runtime/BUILD
	test/BUILD
	tools/codegen/cmd/mixgeninventory/BUILD
	tools/codegen/pkg/inventory/BUILD

no changes added to commit (use "git add" and/or "git commit -a")
```

Then run `make test`, failed for some cases.

```
ERROR: /root/go/src/istio.io/mixer/cmd/server/cmd/BUILD:5:1: GoCompile cmd/server/cmd/~lib~/go_default_library.o failed (Exit 1)
cmd/server/cmd/server.go:222: undefined: "istio.io/mixer/adapter".Inventory2
cmd/server/cmd/server.go:229: undefined: "istio.io/mixer/adapter".Inventory2
2017/08/21 06:08:38 error running compiler: exit status 1
INFO: Elapsed time: 26.757s, Critical Path: 4.88s
FAILED: Build did NOT complete successfully
//pkg/adapter:go_default_test                                            PASSED in 0.1s
//pkg/adapterManager:dispatch_benchmark_test                             PASSED in 0.1s
//pkg/adapterManager:go_default_test                                     PASSED in 0.5s
//pkg/api:go_default_test                                                PASSED in 0.1s
//pkg/aspect:go_default_test                                             PASSED in 0.1s
//pkg/attribute:go_default_test                                          PASSED in 0.7s
//pkg/config/descriptor:go_default_test                                  PASSED in 0.2s
//pkg/config:go_default_test                                             PASSED in 0.3s
//pkg/config/redis:go_default_test                                       PASSED in 0.4s
//pkg/config/store:go_default_test                                       PASSED in 0.3s
//pkg/expr:go_default_test                                               PASSED in 2.1s
//pkg/il/compiler:go_default_test                                        PASSED in 0.1s
//pkg/il/evaluator:go_default_test                                       PASSED in 0.1s
//pkg/il:go_default_test                                                 PASSED in 0.5s
//pkg/il/interpreter:benchmark_tests                                     PASSED in 0.1s
//pkg/il/interpreter:go_default_test                                     PASSED in 0.8s
//pkg/il/text:go_default_test                                            PASSED in 0.3s
//pkg/pool:go_default_test                                               PASSED in 0.7s
//pkg/runtime:go_default_test                                            PASSED in 0.1s
//pkg/status:go_default_test                                             PASSED in 0.3s
//pkg/template:go_default_test                                           PASSED in 0.1s
//pkg/tracing/basic:go_default_test                                      PASSED in 0.4s
//pkg/tracing/zipkin:go_default_test                                     PASSED in 0.7s
//pkg/version:go_default_test                                            PASSED in 1.2s
//template/sample:go_default_test                                        PASSED in 0.1s
//adapter/denier:go_default_test                                      NO STATUS
//adapter/denyChecker:go_default_test                                 NO STATUS
//adapter/genericListChecker:go_default_test                          NO STATUS
//adapter/ipListChecker:go_default_test                               NO STATUS
//adapter/kubernetes:go_default_test                                  NO STATUS
//adapter/list:go_default_test                                        NO STATUS
//adapter/memQuota:go_default_test                                    NO STATUS
//adapter/noop:go_default_test                                        NO STATUS
//adapter/noop2:go_default_test                                       NO STATUS
//adapter/prometheus:go_default_test                                  NO STATUS
//adapter/redisquota:go_default_test                                  NO STATUS
//adapter/serviceControl:go_default_test                              NO STATUS
//adapter/statsd:go_default_test                                      NO STATUS
//adapter/stdioLogger:go_default_test                                 NO STATUS
//cmd/client/cmd:go_default_test                                      NO STATUS
//test:go_default_test                                                NO STATUS
//tools/codegen/pkg/bootstrapgen:go_default_test                      NO STATUS
//tools/codegen/pkg/interfacegen:go_default_test                      NO STATUS
//tools/codegen/pkg/inventory:go_default_test                         NO STATUS
//tools/codegen/pkg/modelgen:go_default_test                          NO STATUS
```

**Comment by gyliu513** [comment]
@geeknoid any comments for this? Thanks.

**Comment by geeknoid** [comment]
I just tried this locally, and it appears as though gazelle decided that the new adapters are useless and took them all out of adapter/BUILD. Gazelle must be getting confused. I don't know anything about gazelle though, not sure who does.

**Comment by jmuk** [comment]
This was fixed by #1124

**Comment by jmuk** [comment]
(by the way I believe this is a bug of Gazelle, reported as https://github.com/bazelbuild/rules_go/issues/744)
